packages/blade/src/components/AutoComplete/_decisions/decisions.md

```diff
@@ -0,0 +1,257 @@
+| labelPosition      | Label Position like other inputs but with new option `inside-input`. This value new value will be added to `AutoComplete` and `SelectInput` | 'top' \| 'left' \| 'inside-input'                               | 'top'             |
+| maxRows            | height restrictions of input ([Checkout "Inactive - Active States" Usage](#with-different-inactive---active-states))                        | 'single' \| 'multiple' \| 'expandable'                          | 'multiple'        |
+| inputValue         | Controlled state of the value inside input                                                                                                  | string                                                          |                   |
+| onInputValueChange | Callback when input value changes                                                                                                           | (inputValue: string) => void                                    |                   |
+| value              | Controlled state of the selected items ([Checkout "Controlled AutoComplete" Usage](#controlled-autocomplete))                               | string[]                                                        |                   |
+| onChange           | Callback when selected items change                                                                                                         | ({ name, values }: { name: string; values: string[] } ) => void |                   |
+
+## Usage
+
+AutoComplete will be built as a new trigger for existing Dropdown component.
+
+Depending on where and how AutoComplete is used, the API can differ. Checkout examples below on how code will look like for a particular usage.
+
+- [Basic Usage](#basic-usage)
+- [Controlled AutoComplete](#controlled-autocomplete)
+- [In BottomSheet](#in-bottomsheet)
+- [With Tags Outside](#with-tags-outside)
+- [With Different Inactive - Active States](#with-different-inactive---active-statess)
+
+### Basic Usage
+
+```jsx
+<Dropdown selectionType="multiple">
+  <AutoComplete label="Select Cities" />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Pune" value="pune" />
+      <ActionListItem title="Hyderabad" value="hyderabad" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>
+```
+
+### Controlled AutoComplete
+
+There are 3 controlled states that are relevant to AutoComplete
+
+1. Controlled state of filtering items
+2. Controlled state of the Input where user types
+3. Controlled state of selected items
+
+```jsx
+const [cities, setCities] = React.useState(allCitiesData);
+const [selections, setSelections] = React.useState([]);
+
+// ...
+
+<AutoComplete
+  // For controlling the selections
+  value={selections}
+  onChange={({ name, values }) => {
+    setSelections(values)
+  }}
+  // For controlling the input value
+  inputValue={}
+  onInputValueChange={(inputValue) => {
```

Comment:
can we pass a promise here instead? We usually use autocomplete when we have to query APIs to fetch data. Passing promises here allows the autocomplete component to abort the previous promise/result whenever input changes before the API call has been resolved. It will relieve the caller from handling such things.

The same promise can also be used for static data

Reply:
Do you mean something like this? this will be possible yes. Let me know if you meant something else.

```js
onInputValueChange={async (inputValue) => {
   const newFilteredData = await fetchData(inputValue);
   setFilteredData(newFilteredData);
}}

// ...

filteredData.map((data) => <ActionList title={data.title} />
```

Reply:
yeah but who does debouncing / aborting of requests when a new query is triggered, would the component handle it or is it left to the user

Reply:
It will be upto the user. This callback will fire on every input change so debounce is something that consumers can add while calling the API.

How much you want to delay abort is contextual to how you're fetching data so we won't have enough context to take call there.

Reply:
The filtering API is changed due to implementation limitations, but I think same thing applies to new API as well so resolving the conversation.